x-pack/plugin/mapper-unsigned-long/src/main/java/org/elasticsearch/xpack/unsignedlong/UnsignedLongFieldMapper.java

```diff
@@ -299,9 +301,13 @@ public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext
 
             if ((operation == FielddataOperation.SEARCH || operation == FielddataOperation.SCRIPT) && hasDocValues()) {
                 return (cache, breakerService) -> {
+                    ValuesSourceType valuesSourceType = metricType == TimeSeriesParams.MetricType.counter
+                        ? TimeSeriesValuesSourceType.COUNTER
+                        : IndexNumericFieldData.NumericType.LONG.getValuesSourceType();
                     final IndexNumericFieldData signedLongValues = new SortedNumericIndexFieldData.Builder(
                         name(),
                         IndexNumericFieldData.NumericType.LONG,
+                        valuesSourceType,
                         (dv, n) -> { throw new UnsupportedOperationException(); }
                     ).build(cache, breakerService);
                     return new UnsignedLongIndexFieldData(signedLongValues, UnsignedLongDocValuesField::new);
```

Comment:
Same question here about the script path.

Reply:
And same answer as above :)

Reply:
Does this one want the `valuesSourceType` passed to it too?

Reply:
Oh yes, good catch, and ScaledFloat too.